src/rpcnet.cpp

```diff
@@ -398,6 +399,34 @@ static UniValue GetNetworksInfo()
     return networks;
 }
 
+UniValue getdeprecationinfo(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() != 0)
+        throw runtime_error(
+            "getdeprecationinfo\n"
+            "Returns an object containing current version and deprecation block height.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"version\": xxxxx,                      (numeric) the server version\n"
+            "  \"subversion\": \"/MagicBean:x.y.z[-v]/\",     (string) the server subversion string\n"
```

Comment:
I find the naming of this irritating (it's an extended representation of the version), but it matches existing Bitcoin-inherited RPCs and API designs, so leave it as-is.